server/routes/altSearch.ts

```diff
@@ -0,0 +1,214 @@
+  const router = Router()
+
+  const get = (path: string | string[], handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
+  const post = (path: string | string[], ...handlers: RequestHandler[]) =>
+    router.post(
+      path,
+      handlers.map(handler => asyncMiddleware(handler)),
+    )
+
+  get(['/prisoner'], (req, res) => {
+    const search = req?.body?.search
+
+    res.render('pages/alt-search/prisoner', {
+      search,
+      visit: req.originalUrl.includes('-visit'),
+      showEstablishmentSwitcher: true,
+    })
+  })
+
+  post(['/prisoner'], body('search').trim('. '), (req, res) => {
+    const { search } = req.body
+
+    return res.redirect(
+      url.format({
+        pathname: `/altSearch/prisoner/results`,
+        query: {
+          ...(search && { search }),
+        },
+      }),
+    )
+  })
+
+  get(['/prisoner/results', '/prisoner-visit/results'], async (req, res) => {
+    const { prisonId } = req.session.selectedEstablishment
+    const search = typeof req.query.search === 'string' ? req.query.search : ''
+
+    const currentPage = typeof req.query.page === 'string' ? req.query.page : ''
+    const parsedPage = Number.parseInt(currentPage, 10) || 1
+    const { pageSize } = config.apis.prisonerSearch
+
+    const validationErrors = validatePrisonerSearch(search)
+    const errors = validationErrors ? [validationErrors] : []
+    const hasValidationErrors = errors.length > 0
+
+    const { results, numberOfPages, numberOfResults, next, previous } = hasValidationErrors
+      ? { results: 0, numberOfPages: 0, numberOfResults: 0, next: 0, previous: 0 }
+      : await prisonerSearchService.getPrisoners(search, prisonId, res.locals.user.username, parsedPage, false) // todo
+
+    if (!hasValidationErrors) {
+      await auditService.prisonerSearch({
+        searchTerms: search,
+        prisonId,
+        username: res.locals.user.username,
+        operationId: res.locals.appInsightsOperationId,
+      })
+    }
+
+    const currentPageMax = parsedPage * pageSize
+    const to = numberOfResults < currentPageMax ? numberOfResults : currentPageMax
+
+    const pageLinks = getResultsPagingLinks({
+      pagesToShow: config.apis.prisonerSearch.pagesLinksToShow,
+      numberOfPages,
+      currentPage: parsedPage,
+      searchParam: `search=${search}`,
+      searchUrl: `/search/prisoner/results`,
+    })
+
+    const validPrisonerNumber = extractPrisonerNumber(search)
+    let prisonerNotFoundMessage
+    if (numberOfResults === 0 && validPrisonerNumber) {
+      prisonerNotFoundMessage = await prisonerSearchService.getPrisonerNotFoundMessage(
+        validPrisonerNumber,
+        req.session.selectedEstablishment.prisonName,
+        res.locals.user.username,
+      )
+    } else {
+      prisonerNotFoundMessage = `There are no results for this name or number at ${req.session.selectedEstablishment.prisonName}.`
+    }
+
+    res.render('pages/alt-search/prisonerResults', {
```

Comment:
Again standards. URLs are lowercase with dashes but everything else is camelcase, so here maybe should be `altSearch`